services/scrutinizer/scrutinizer-build.service.js

```diff
@@ -0,0 +1,131 @@
+'use strict'
+
+const Joi = require('joi')
+const { isBuildStatus, renderBuildStatusBadge } = require('../build-status')
+const ScrutinizerBase = require('./scrutinizer-base')
+
+const schema = Joi.object({
+  default_branch: Joi.string().required(),
+  applications: Joi.object()
+    .pattern(
+      /^/,
+      Joi.object({
+        build_status: Joi.object({
+          status: Joi.alternatives().try(isBuildStatus, Joi.equal('unknown')),
+        }).required(),
+      })
+    )
+    .required(),
+}).required()
+
+class ScrutinizerBuildBase extends ScrutinizerBase {
+  static get category() {
+    return 'build'
+  }
+
+  static get defaultBadgeData() {
+    return {
+      label: 'build',
+    }
+  }
+
+  transform({ json, branch }) {
+    branch = this.transformBranch({ json, wantedBranch: branch })
```

Comment:
Maybe call this `branchInfo`? It's a bit confusing to re-use the name since (old) `branch` is a string and (new) `branch` is a nested object.

Or destructure:

```js
const { build_status: { status } } = this.transformBranch(...)
return { status }
```

You could also inline this in `render()` to reduce indirection.

Reply:
> Maybe call this branchInfo? It's a bit confusing to re-use the name since (old) branch is a string and (new) branch is a nested object.

> Or destructure:

Good point!

> You could also inline this in render() to reduce indirection

Not sure I follow as there's no `render` functions here. Are you suggesting handling all this below by skipping `transform` and inlining in  `renderBuildStatusBadge` call? Something else?

Reply:
Oops, I meant inlining the `transform()` function in _`handle()`_.

Reply:
Updated, let me know if that looks any better